File: ai/src/main/java/org/pingel/gestalt/core/refactor/RenameTransformRefactoring.java
package org.pingel.gestalt.core.refactor;

import org.pingel.gestalt.core.Lexicon;
import org.pingel.gestalt.core.Name;

public class RenameTransformRefactoring extends Refactoring {

	Name from;
	Name to;
	
	public RenameTransformRefactoring(Lexicon lexicon, Name from, Name to)
	{
		super(lexicon);
		this.from = from;
		this.to = to;
	}

	public void execute()
	{
		lexicon.renameTransform(from, to);
	}
	
}

File: gestalt/src/main/scala/org/pingel/gestalt/core/refactor/RenameTransformRefactoring.scala
package org.pingel.gestalt.core.refactor

import org.pingel.gestalt.core.Lexicon
import org.pingel.gestalt.core.Name

case class RenameTransformRefactoring(lexicon: Lexicon, from: Name, to: Name)
extends Refactoring(lexicon) 
{
	def execute(): Unit = {
		lexicon.renameTransform(from, to)
	}
	
}
